### ISSUE TITLE
Add support in Backend for multiple loss functions and optimizers

### ISSUE THREAD

**Issue by pmasilva20** [Dioben/PI18#97]
See both links:
https://www.tensorflow.org/api_docs/python/tf/keras/optimizers
https://www.tensorflow.org/api_docs/python/tf/keras/losses
